main.tf

```diff
@@ -1 +1,371 @@
-# Your Terraform code goes here :)
+locals {
+  bucket_name                     = "${var.prefix}${random_id.uniq.hex}"
+  bucket_region                   = length(var.bucket_region) > 0 ? var.bucket_region : data.aws_region.current.name
+  mfa_delete                      = var.bucket_enable_versioning && var.bucket_enable_mfa_delete ? "Enabled" : "Disabled"
+  bucket_enable_versioning        = var.bucket_enable_versioning ? "Enabled" : "Suspended"
+  cross_account_policy_name       = "${var.prefix}-cross-acct-policy-${random_id.uniq.hex}"
+  iam_role_arn                    = module.lacework_eks_audit_iam_role.arn
+  iam_role_external_id            = module.lacework_eks_audit_iam_role.external_id
+  cross_account_iam_role_name     = "${var.prefix}-ca-${random_id.uniq.hex}"
+  sns_name                        = "${var.prefix}${random_id.uniq.hex}"
+  firehose_iam_role_name          = "${var.prefix}-fh-${random_id.uniq.hex}"
+  firehose_policy_name            = "${var.prefix}-fh-policy-${random_id.uniq.hex}"
+  firehose_delivery_stream_name   = "${var.prefix}${random_id.uniq.hex}"
+  eks_cw_iam_role_name            = "${var.prefix}-cw-${random_id.uniq.hex}"
+  cw_iam_policy_name              = "${var.prefix}-cw-policy-${random_id.uniq.hex}"
+  cloudwatch_permission_resources = "arn:aws:logs:*:${data.aws_caller_identity.current.account_id}:log-group:/aws/eks/*:*"
+  log_region                      = "logs.${var.cloudwatch_region}.amazonaws.com"
+}
+
+provider "aws" {
+  region = var.cloudwatch_region
+}
+
+resource "random_id" "uniq" {
+  byte_length = 4
+}
+
+resource "aws_sns_topic" "eks_sns_topic" {
+  name = local.sns_name
+}
+
+resource "aws_sns_topic_policy" "eks_sns_topic_policy" {
+  arn = aws_sns_topic.eks_sns_topic.arn
+
+  policy = data.aws_iam_policy_document.eks_sns_topic_policy.json
+}
+
+# we need the identity of the caller to get their account_id
+data "aws_caller_identity" "current" {}
+
+# this allows us to get the region configured in the aws provider
+data "aws_region" "current" {}
+
+data "aws_iam_policy_document" "eks_sns_topic_policy" {
+
+  statement {
+    sid    = "AllowLaceworkToSubscribe"
+    effect = "Allow"
+    principals {
+      identifiers = ["arn:aws:iam::${var.lacework_aws_account_id}:root"]
```

Comment:
I'm just noticing our platform supports US Gov AWS elsewhere, does this setup also need to support it? This ARN is hardcoded for non-Gov

Reply:
That's a fair point. I'm not aware of this requirement at the moment. If it is required we can always re-work the implementation in the future.